### PR TITLE
Infrastructure redundancy

### PR DESCRIPTION
**Pull request by sindkor**
Addressing #7. Adds the support for availability set. The user can now chose one of two options:

- No infrastructure redundancy
- Having the module deploy an availability set, number of fault- and update domains can be specified.

For reasons of backwards-compatibility, no infrastructure redundancy is the default for now, but his might change in a later release. I've also added a small enhancement that gives the option to set a user-facing friendly name to the workspace.

 - [x] Redundancy options implemented
 - [x] User-facing workspace name
 - [x] Functionality tested
 - [x] Examples added

### CODE REVIEW

File: compute.tf [hunk 1]
```diff
@@ -9,10 +9,12 @@ resource "azurerm_windows_virtual_machine" "main" {
   admin_username        = var.local_admin_username
   admin_password        = local.local_admin_password[count.index]
 
+  zone                = var.use_availability_zones ? count.index % 3 + 1 : null
```
Comment (nikolaifa) [inline compute.tf:12]: There is a risk that multiple VMs within the same zone will be deployed within the same fault and/or update domain. I suggest the module instead manages an availability set.

Comment (sindkor) [reply compute.tf:12]: Are you sure? I don't think it makes sense to talk about fault domain when we aren't using availability sets. My understanding is that a fault domain is a grouping of VMs within an availability set that shares power and network switching - essentially a rack.
>Fault domains define the group of virtual machines that share a common power source and network switch.

https://docs.microsoft.com/en-us/azure/virtual-machines/availability-set-overview

Whereas an availability zone is essentially its own datacenter:
>Each zone is composed of one or more datacenters equipped with independent power, cooling, and networking infrastructure.

https://docs.microsoft.com/en-us/azure/availability-zones/az-overview

Comment (nikolaifa) [reply compute.tf:12]: My perspective is that using availability zones is the better pattern as it gives control of the availability settings. 
It makes more sense to manage the availability set within the module as the module itself compromises all the hosts that will run the "application".

Regarding the noise, I suggest setting default values for `platform_update_domain_count` and `platform_update_domain_count`. Its your decision whether that constitutes as too much noise or not.

**Review by nikolaifa — APPROVED** [review]
LGTM

**Pull request merged**